fml/ascii_trie.cc

```diff
@@ -0,0 +1,68 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+#include "flutter/fml/ascii_trie.h"
+
+namespace fml {
+typedef AsciiTrie::TrieNode TrieNode;
+
+namespace {
+void Add(TrieNode** trie, const char* entry) {
+  int ch = entry[0];
+  assert(ch < 128);
+  if (ch != 0) {
+    if (!*trie) {
+      TrieNode* newNode = (TrieNode*)calloc(sizeof(TrieNode), 1);
```

Comment:
Not sure why the placement new is necessary. Please just use standard C++ idioms.

Reply:
`calloc` is used because it is faster if you want a zeroed out allocation.  In C++ you'd need a custom allocator to do something similar.  This is a private implementation detail of the AsciiTrie, it's external API follows C++ conventions.

Reply:
On ownership and pointers: go/fluttercpp#Ownership_and_Smart_Pointers